pkg/icingadb/ha.go

```diff
@@ -323,6 +334,32 @@ func (h *HA) realize(ctx context.Context, s *icingaredisv1.IcingaStatus, t *type
 	return nil
 }
 
+func (h *HA) realizeLostHeartbeat() {
+	_ = retry.WithBackoff(
```

Comment:
Okay, so if you insist on keeping this function, let's consider it in more detail: what's the reason for using `retry.WithBackoff()` here? There are multiple other functions in this file doing SQL queries and the only other one using it is `realize()`, but as far as I can tell, it's used there because it's using a transaction with isolation level serializable that may fail to commit and this isn't the case here.

Also, this is some kind of "let's try to be nice" feature. The worst that would happen if that query fails would be that the next instance to take over due to an expired heartbeat would reset the `responsible` column for the old instance (similar to when it just crashed instead).

Changing this even removes most of the duplication with `realize()` so I wouldn't even be that unhappy about keeping it then I guess.

Reply:
The reason is... a copy&🐘 error.

(Btw. I don’t _insist_. I was just reasoning.)